app/views/lookup_keys/_fields.html.erb

```diff
@@ -1,60 +1,61 @@
-
-  <%= remove_child_link(_("Remove %s?") % (f.object.new_record? ? _("Variable") : f.object), f , {:class => 'btn btn-danger hide'}) unless controller_name == "lookup_keys" %>
-  <%= text_f f, :key, :disabled => f.object.is_param, :size => "col-md-8"  %>
-  <%= f.hidden_field :key if is_param %>
-  <%= textarea_f f, :description, :rows => :auto, :size => "col-md-8"  %>
-
-  <%= show_puppet_class f %>
-  <%= checkbox_f(f, :override, :onchange => 'toggleOverrideValue(this)', :size => "col-md-8",
-                 :help_block => _('Whether the smart variable value is managed by Foreman')
-      ) if is_param%>
-
-  <%= param_type_selector(f, :onchange => 'keyTypeChange(this)')  %>
-  <%= checkbox_f(f, :use_puppet_default, :label => _('Use Puppet default'), :size => "col-md-8",
-                    :help_block => use_puppet_default_help,
-                    :onchange=>'toggleUsePuppetDefaultValue(this, "default_value")',
-                    :disabled => (f.object.is_param && !f.object.override)) if is_param %>
-  <%= textarea_f f, :default_value, :value => f.object.default_value_before_type_cast,:size => "col-md-8",
-                    :disabled => (f.object.is_param && (!f.object.override || f.object.use_puppet_default)),
-                    :fullscreen => :true,
-                    :rows => :auto, :help_block => _("Value to use when there is no match") %>
-  </br>
-  <div <%= "id=#{(f.object.key || 'new_lookup_keys').to_s.gsub(' ','_')}_lookup_key_override_value" %> style=<%= "display:none;" if (f.object.is_param && !f.object.override) %>>
-    <legend><%= _("Optional input validator") %></legend>
-    <p class="help-block">
-      <%= _('Note that if you use ERB as a value of parameter, value will be validated during ENC evaluation. If value is invalid, ENC evaluation will fail.') %>
-    </p>
-    <%= checkbox_f(f, :required, :size => "col-md-8", :disabled => !f.object.override,
-                      :help_block => _("If checked, will raise an error if there is no default value and no matcher provide a value.")
+  <%= remove_child_link(_("Remove %s?") % (f.object.new_record? ? _("Variable") : f.object), f, { :class => 'btn btn-danger hide' }) unless controller_name == "lookup_keys" %>
+  <fieldset>
+    <legend><%= _("Parameter Details") %></legend>
+    <%= text_f f, :key, :disabled => f.object.is_param, :size => "col-md-8" %>
+    <%= f.hidden_field :key if is_param %>
+    <%= textarea_f f, :description, :rows => :auto, :size => "col-md-8" %>
+    <%= text_f(f, :environment_classes, :value => f.object.environment_classes.map(&:environment).to_sentence, :label => _('Puppet Environments'), :size => "col-md-8", :disabled => true) if is_param %>
+    <%= show_puppet_class f %>
+  </fieldset>
+  <fieldset>
+    <legend><%= _("Add Override") %></legend>
+    <h6><%= _("Override the Puppet class default value") %></h6>
```

Comment:
I'd suggest expanding this a bit to explain what the section does - no more than one or two sentences.